modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpChannel.java

```diff
@@ -134,11 +134,13 @@ public void sendResponse(RestResponse response) {
                 promise.addListener(ChannelFutureListener.CLOSE);
             }
 
+            final Object msg;
             if (pipelinedRequest != null) {
-                channel.writeAndFlush(pipelinedRequest.createHttpResponse(resp, promise));
+                msg = pipelinedRequest.createHttpResponse(resp);
             } else {
-                channel.writeAndFlush(resp, promise);
+                msg = resp;
             }
+            channel.writeAndFlush(msg, promise);
```

Comment:
I think the release flag should be move after this? if we don't use the promise, it's useless no?